providers/dns/hurricanedns/hurricanedns.go

```diff
@@ -0,0 +1,221 @@
+// Package hurricanedns implements a DNS provider for solving the DNS-01 challenge using Hurricane Electric DNS.
+package hurricanedns
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/xenolf/lego/acme"
+	"github.com/PuerkitoBio/goquery"
+	"net/url"
+	"regexp"
+)
+
+// DNSProvider is an implementation of the acme.ChallengeProvider interface that uses
+// dns.he.net web page to manage TXT records for a domain.
+type DNSProvider struct {
+	heBaseURL  string
+	domainName string
+	userName   string
+	password   string
+}
+
+// NewDNSProvider returns a DNSProvider instance configured for Hurricane Electric DNS.
+// Credentials must be passed in the environment variables: HE_DOMAIN_NAME,
+// HE_USERNAME, HE_PASSWORD.
+func NewDNSProvider() (*DNSProvider, error) {
+	domainName := os.Getenv("HE_DOMAIN_NAME")
+	userName := os.Getenv("HE_USERNAME")
+	password := os.Getenv("HE_PASSWORD")
+	return NewDNSProviderCredentials(domainName, userName, password)
```

Comment:
Could rewrite like the following example:
https://github.com/xenolf/lego/blob/c4bbb4b819c087d02066fe745d7ca5b60a480d36/providers/dns/cloudflare/cloudflare.go#L32-L39